lib/apiservers/engine/backends/container.go

```diff
@@ -1315,6 +1316,8 @@ func (c *Container) ContainerChanges(name string) ([]docker.Change, error) {
 			change.Kind = docker.ChangeAdd
 		case "D":
 			change.Kind = docker.ChangeDelete
+			p := strings.TrimPrefix(filepath.Base(change.Path), docker.WhiteoutPrefix)
+			change.Path = filepath.Join(filepath.Dir(change.Path), p)
```

Comment:
oh wait, this is Delete case already so it is guaranteed to have WhiteoutPrefix in it. 

LGTM